test/session_tests/DocumentQueryTest.ts

```diff
@@ -8,220 +8,493 @@ import {RequestExecutor} from "../../src/Http/Request/RequestExecutor";
-        documentType: Product
-      }).whereEquals<string>('name', 'test101').get();
-        
-      expect(results[0].name).to.equals('test101');
-    });
-
-    it('should query by double index joined by "OR" operator', async () => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product
-      })
-      .whereEquals<string>('name', 'test101')
-      .whereEquals<number>('uid', 4).get();
-        
+
+    it('should query by exact', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .exact<string>('name', 'withnesting').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.map(function (obj) {
+        expect(obj).to.include({'name': 'withNesting'});
+      });
+
+    });
+
+    it('should query with whereEqualsAndOr', async () => {
+      const results: Universal[] = await  store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .whereEqualsAndOr<number>('name','withNesting','order',3,'order',4).get();
+
       expect(results).to.have.lengthOf(2);
+
+      results.map(function(obj) {
+
+        expect(obj).to.include({'name': 'withNesting'});
+
+      });
+
     });
 
-    it('should query by double index joined by "AND" operator', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product,
-        waitForNonStaleResults: true,
-        usingDefaultOperator: QueryOperators.AND
-      })
-      .whereEquals<string>('name', 'test107')
-      .whereEquals<number>('uid', 5).get();
+    it('should query with orderBy as long', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .orderBy<string>('order as long', 'ASC').get();
+
+      expect(results).to.have.lengthOf(10);
+
+        results.map(function(obj, index) {
+          index += 1;
+          expect(obj.order).to.be.equal(index);
+        });
 
-      expect(results).to.have.lengthOf(1);
     });
 
-    it('should query by whereIn', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product
-      }).whereIn<string>('name', ['test101', 'test107', 'test106']).get();
-      
-      expect(results).to.have.lengthOf(4);      
+    it('should query with orderBy as string', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .orderBy<string>('order', 'ASC').get();
+
+      expect(results[1].order).to.be.equal(10);
```

Comment:
When testing `orderBy` method you should iterate over the list of results and check whether they are ordered.